File: pending/src/main/java/io/mendirl/demo/pending/repository/PendingRepository.java
package io.mendirl.demo.pending.repository;

import io.hypersistence.utils.spring.repository.BaseJpaRepository;
import io.mendirl.demo.pending.entity.PendingEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface PendingRepository extends BaseJpaRepository<PendingEntity, Long> {

//    @Override
//    PendingEntity getReferenceById(Long id);
//
//    @Override
//    PendingEntity saveAndFlush(PendingEntity pending);

}

File: kpending/src/main/kotlin/io/mendirl/demo/pending/repository/PendingRepository.kt
package io.mendirl.demo.pending.repository

import io.mendirl.demo.pending.entity.PendingEntity
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PendingRepository: JpaRepository<PendingEntity, Long> {

    override fun getReferenceById(id: Long): PendingEntity

    override fun <S : PendingEntity?> saveAndFlush(entity: S & Any): S & Any

}
